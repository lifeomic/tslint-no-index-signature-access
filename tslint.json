{
  "extends": "tslint-config-semistandard",
  "rules": {
    "space-before-function-paren": false
  }
}
