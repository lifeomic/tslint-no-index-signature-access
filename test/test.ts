import { exec } from 'child_process';
import { join } from 'path';
import * as Lint from 'tslint';

const tslintBin = join('..', 'node_modules', '.bin', 'tslint');
const tslintConfig = join('.', 'tslint.json');
const tslintFormat = 'json';
const tsFiles = `${join('.', '*.ts')} ${join('.', '*/*.ts')}`;
const tsconfig = join('.', 'tsconfig.json');

test('all cases pass', async () => {
  const [stdout, stderr] = await new Promise<[string, string]>(function(
    resolve
  ) {
    exec(
      `${tslintBin} -p ${tsconfig} -c ${tslintConfig} -r .. -t ${tslintFormat} ${tsFiles}`,
      { cwd: __dirname },
      (_, stdout, stderr) => {
        // Intentionally ignore errors because the linter is expected to fail
        resolve([stdout, stderr]);
      }
    );
  });

  expect(stderr).toBe('');

  // Only validate failures and names.
  const actual = (JSON.parse(stdout) as Lint.IRuleFailureJson[]).map((x) => ({
    failure: x.failure,
    name: x.name.replace(/.*\/test\//, ''),
    startPosition: x.startPosition
  }));

  expect(actual).toMatchSnapshot();
});
