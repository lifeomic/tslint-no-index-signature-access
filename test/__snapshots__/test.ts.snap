// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`all cases pass 1`] = `
Array [
  Object {
    "failure": "References to generic properties on TypeWithIndex are not allowed",
    "name": "case-class-subclass.ts",
    "startPosition": Object {
      "character": 16,
      "line": 8,
      "position": 191,
    },
  },
  Object {
    "failure": "References to generic properties on TypeWithIndex are not allowed",
    "name": "case-class.ts",
    "startPosition": Object {
      "character": 11,
      "line": 6,
      "position": 137,
    },
  },
  Object {
    "failure": "References to generic properties on TypeWithIndex are not allowed",
    "name": "case-class.ts",
    "startPosition": Object {
      "character": 11,
      "line": 6,
      "position": 137,
    },
  },
  Object {
    "failure": "References to generic properties on TypeWithIndex are not allowed",
    "name": "case-interface-extends.ts",
    "startPosition": Object {
      "character": 0,
      "line": 8,
      "position": 230,
    },
  },
  Object {
    "failure": "References to generic properties on TypeWithIndex are not allowed",
    "name": "case-interface.ts",
    "startPosition": Object {
      "character": 0,
      "line": 6,
      "position": 157,
    },
  },
  Object {
    "failure": "References to generic properties on TypeWithIndex are not allowed",
    "name": "case-interface.ts",
    "startPosition": Object {
      "character": 0,
      "line": 6,
      "position": 157,
    },
  },
  Object {
    "failure": "References to generic properties on TypeWithIndex are not allowed",
    "name": "case-type-alias-union.ts",
    "startPosition": Object {
      "character": 11,
      "line": 10,
      "position": 216,
    },
  },
  Object {
    "failure": "References to generic properties on TypeWithIndex are not allowed",
    "name": "case-type-alias.ts",
    "startPosition": Object {
      "character": 11,
      "line": 6,
      "position": 140,
    },
  },
  Object {
    "failure": "References to generic properties on TypeWithIndex are not allowed",
    "name": "case-type-alias.ts",
    "startPosition": Object {
      "character": 11,
      "line": 6,
      "position": 140,
    },
  },
]
`;
